packages/stream-json/writer.js

```diff
@@ -0,0 +1,27 @@
+// @ts-check
+/// <reference types="ses"/>
+
+/* Adapts a Writer<Uint8Array> to a JSON Writer<any>, where reader and writer
+ * streams are modeled as hybrid async iterators + generators.
+ */
+const textEncoder = new TextEncoder();
+
+/**
+ * @param {import('@endo/stream').Writer<Uint8Array, void>} writer
+ */
+export const makeJsonWriter = writer => {
```

Comment:
Any reason not to write this as generator like reader?

Reply:
Yeah, theoretically there are two kinds of generators: even and odd. An odd generator starts at the top of the function and even generator would advance immediately to the first yield. We only have odd generators, which means that the first value sent to a generator gets discarded. It’s possible to compensate by using a wrapped generator and inducing `next()` immediately, but the type is hard to pin down.

Reply:
The old `function.next` tc39 proposal died because no one was willing to invest to move it forward. That would have allowed generators (and async generators) to read the argument of the first `next` call. Oh well.

I'll use this as an opportunity to lobby, separately, for an updater adaptor which would also rename the API surface. But these PRs should not block on that debate.